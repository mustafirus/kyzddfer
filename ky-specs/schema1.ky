rack ver(1.0)
  tables
    users !master display_name(Users), description(System users table)
      # 'id' є неявним полем згідно зі специфікацією
      name varchar(100) !required label(User Name)
      email varchar(255) !unique label(Email Address)

    posts display_name(Posts)
      # 'id' є неявним полем
      user_id ref(users) !fk label(Author)
      title varchar(255)
      content text

  apps
    main_app version(1.0)
      # Визначення леяуту: рядок метаданих та рядок кореневого вузла
      # знаходяться на одному рівні вкладеності.
      posts_list
      list table(posts)
        # Дочірні елементи списку - це прості вузли, що представляють поля.
        id label(ID)
        title label(Title)
        user_id.name label(Author Name)
