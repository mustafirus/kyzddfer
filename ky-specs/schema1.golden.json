{
  "rack": {
    "apps": [
      {
        "name": "main_app",
        "attrs": {
          "version": "1.0"
        },
        "layouts": [
          {
            "name": "posts_list",
            "root_node": {
              "tag": "list",
              "attrs": {
                "table": "posts"
              },
              "nodes": [
                {
                  "tag": "id",
                  "attrs": {
                    "label": "ID"
                  }
                },
                {
                  "tag": "title",
                  "attrs": {
                    "label": "Title"
                  }
                },
                {
                  "tag": "user_id.name",
                  "attrs": {
                    "label": "Author Name"
                  }
                }
              ]
            }
          }
        ]
      }
    ],
    "tables": [
      {
        "name": "posts",
        "attrs": {
          "display_name": "Posts"
        },
        "fields": [
          {
            "name": "content",
            "type": "text"
          },
          {
            "name": "title",
            "type": "varchar(255)"
          },
          {
            "name": "user_id",
            "type": "ref(users)",
            "attrs": {
              "label": "Author"
            },
            "flags": [
              "fk"
            ]
          }
        ]
      },
      {
        "name": "users",
        "attrs": {
          "description": "System users table",
          "display_name": "Users"
        },
        "flags": [
          "master"
        ],
        "fields": [
          {
            "name": "email",
            "type": "varchar(255)",
            "attrs": {
              "label": "Email Address"
            },
            "flags": [
              "unique"
            ]
          },
          {
            "name": "name",
            "type": "varchar(100)",
            "attrs": {
              "label": "User Name"
            },
            "flags": [
              "required"
            ]
          }
        ]
      }
    ]
  }
}